packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/QualitySelectorHostApiImpl.java

```diff
@@ -38,7 +38,8 @@ public static class QualitySelectorProxy {
         @Nullable FallbackStrategy fallbackStrategy) {
       // Convert each index of VideoQualityConstraint to Quality.
       List<Quality> qualityList = new ArrayList<Quality>();
-      for (Long qualityIndex : videoQualityConstraintIndexList) {
+      for (int i = 0; i < videoQualityConstraintIndexList.size(); i++) {
+        int qualityIndex = ((Number) videoQualityConstraintIndexList.get(i)).intValue();
```

Comment:
This is a fix for Long -> Integer casting issues that the integration test revealed.

Reply:
Using your former suggestion, this is an example of the error I got: https://firebase.corp.google.com/project/flutter-infra-staging/testlab/histories/bh.90e767c88242d274/matrices/6437969997948729501/executions/bs.a725e88a067fbf4d/testcases/2/test-cases.

I'm open to suggestions, but I could not find a fix and after investigating, it's unclear why I'm getting the error and only in Firebase like that. May be something with Java versions but unsure.

Reply:
Turns out this is an issue because I was trying to index an array with a Long, most likely. After talking to @reidbaker, I'm going to take his suggestion and rework how I send a`Quality` from Dart to Java rather than trying to send indices of enums that represent `Quality`s.